app/clients/hedera/client.go

```diff
@@ -125,50 +124,69 @@ func (hc Node) SubmitScheduledTokenBurnTransaction(tokenID hedera.TokenID, amoun
 // SubmitTopicConsensusMessage submits the provided message bytes to the
 // specified HCS `topicId`
 func (hc Node) SubmitTopicConsensusMessage(topicId hedera.TopicID, message []byte) (*hedera.TransactionID, error) {
-	tx, err := hedera.NewTopicMessageSubmitTransaction().
-		SetTopicID(topicId).
-		SetMessage(message).
-		SetMaxRetry(hc.maxRetry).
-		FreezeWith(hc.client)
+	attempt := 0
+	for attempt <= hc.maxRetry {
+		attempt++
+		tx, err := hedera.NewTopicMessageSubmitTransaction().
+			SetTopicID(topicId).
+			SetMessage(message).
+			SetMaxRetry(hc.maxRetry).
+			FreezeWith(hc.client)
 
-	if err != nil {
-		return nil, err
-	}
+		if err != nil {
+			return nil, err
+		}
 
-	hc.logger.Debugf("Submit Topic Consensus Message, with transaction ID [%s] and Node Account IDs: %v",
-		tx.GetTransactionID(),
-		tx.GetNodeAccountIDs(),
-	)
+		hc.logger.Debugf("Submit Topic Consensus Message, with transaction ID [%s] and Node Account IDs: %v",
+			tx.GetTransactionID(),
+			tx.GetNodeAccountIDs(),
+		)
```

Comment:
nit: I think we need to log the number of retry, otherwise we may have the same logged line, which could be confusing